### PR TITLE
US130990 - Add support for reading context object from the document root

### PR DESCRIPTION
**Pull request by grant-cleary**
Because we're fetching more complicated checks (outside of a feature flag), we need to be able to pass more information into the component than we did before. This PR hooks up the context look-up we need to do and adjusts the existing renderers to extract the information they need out of the context object.

Because we also need the host in the case of FRAs (specifically for @mentions URL resolution), we unfortunately still do need ifrau for this. :(

### CODE REVIEW

File: components/html-block/html-block.js
Comment (grant-cleary) [inline components/html-block/html-block.js:12]: I recently added a new feature to ifrau to sync html element data attributes automatically. For FRAs to take advantage of this particular feature within this component, they'll need to update ifrau to v0.36.0.

File: components/html-block/html-block.js
Comment (grant-cleary) [inline components/html-block/html-block.js:15]: I wish I could not use ifrau at all, but @mentions requires links to profile cards within the LMS, which are unfortunately LMS routes and thus within FRAs definitely need to have the host pre-pended.

Right now @mentions isn't available in any FRAs, but that may change.

File: helpers/mathjax.js
Comment (grant-cleary) [inline helpers/mathjax.js:29]: This has been supported in the LMS for some time now; it just wasn't propagated here originally.

File: components/html-block/html-block.js
Comment (grant-cleary) [reply components/html-block/html-block.js:12]: Also note: This context object isn't set in the LMS yet. I'm just trying to lay down the groundwork in the components first.

**Comment by grant-cleary** [comment]
Crap, forgot to run linting. 🤦

File: components/html-block/html-block.js
Comment (grant-cleary) [reply components/html-block/html-block.js:27]: One thing maybe worth considering is that if we slapped the `valenceHost` as a data attribute on the host, we could use the new marshaling to pull that into the iframe without needing to explicitly invoke the client.

File: components/html-block/html-block.js
Comment (dbatiste) [inline components/html-block/html-block.js:10]: I think there is a "slim" client that we should be looking into using for our core components that require such a check.  I don't think it's documented though, so I think we might need to look into the code a little to figure out what exactly it does, but I think it is meant to avoid some of the issues we've had with client's clobbering each other.

File: components/html-block/html-block.js
Comment (dbatiste) [reply components/html-block/html-block.js:24]: What does this yield if the data attribute doesn't exist?  Should we be defaulting to an empty object?

File: components/html-block/html-block.js
Comment (dbatiste) [inline components/html-block/html-block.js:23]: Seems like we should be able to avoid this by moving the ifrau bit out of this promise handler (i.e. move it above).

File: components/html-block/html-block.js
Comment (dbatiste) [inline components/html-block/html-block.js:192]: Hmmmm. I think it would be preferred if the renderer can get the context themselves, as opposed to depending on html-block passing in an object that contains everything it needs.  Was this something that was considered?

Comment (grant-cleary) [reply components/html-block/html-block.js:192]: Yeah, I didn't have any strong reason for doing it this way. It felt slightly cleaner to centralize things, but I figured you would have ideas on this when you got back, since I don't really know the long-term vision for this. I can easily make this change. :)

File: components/html-block/html-block.js
Comment (grant-cleary) [reply components/html-block/html-block.js:10]: Yeah, that's a good point. If we're only fetching the host we should be able to use that.

File: components/html-block/html-block.js
Comment (grant-cleary) [reply components/html-block/html-block.js:24]: Yeah, I haven't updated this PR since I made the changes to the HTML editor context stuff, but I was intending on propagating those changes here (or wherever we move the context fetching).

File: components/html-block/html-block.js
Comment (grant-cleary) [reply components/html-block/html-block.js:23]: Oh, you're right. I can do that in the editor too I think.

File: helpers/mathjax.js
Comment (grant-cleary) [inline helpers/mathjax.js:4]: This provider hasn't been created yet, so an LMS PR will be coming to handle that shortly. This also introduces an implicit dependency on FRAs - specifically, FRAs who want this functionality will need to update ifrau to at least 0.36.0 or they won't be able to take advantage of the `data-` attribute syncing.

**Review by dbatiste — APPROVED** [review]

**Review by dmd2l — APPROVED** [review]

File: helpers/framed.js
Comment (grant-cleary) [inline helpers/framed.js:24]: Ran into this while testing. Accidentally inverted this logic when I switched this from `!evt.data.isFramed` to an explicit `undefined` check. 😬

Comment (dbatiste) [reply helpers/framed.js:24]: Maybe this was intended to be `if (!evt || !evt.data || !evt.data.isFramed)`?

Comment (dbatiste) [reply helpers/framed.js:24]: Oh, that's what it used to be.  Was the explicit check needed?

Comment (grant-cleary) [reply helpers/framed.js:24]: No, it just felt... More correct? The host page will only ever return `true` or `undefined`, but it still feels more correct to explicitly resolve to `false` if `isFramed` happens to be false somehow.

Comment (dbatiste) [reply helpers/framed.js:24]: Sometimes I write it to be specific, but other times not, to be more tolerant of both `undefined` and `null` possible values.  It's not an issue as long a `null` is never a possible value, so more just wondering.

**Review by dbatiste — APPROVED** [review]

File: helpers/framed.js
Comment (grant-cleary) [reply helpers/framed.js:24]: Fair enough. Yeah, this was a bit of a weird one.

**Comment by grant-cleary** [comment]
Going to hold off on merging this until the LMS providers are merged; otherwise, this could break LaTeX rendering in other components that may be consuming the HTML block component.

**Pull request merged**
